docs/design/storagenode-downtime-tracking.md

```diff
@@ -0,0 +1,221 @@
+# Storage Node Downtime Tracking
+
+## Abstract
+
+The goal is tracking Storage Nodes downtime per Satellite.
+
+## Background
+
+The [Disqualification design document](disqualification.md) describes when the Storage Nodes get disqualified based on the reputation scores described in the [Node Selection design document](node-selection.md).
+
+The current disqualification showed to us that several Storage Nodes got disqualified because of their uptime stats without giving clear and fair evidence of it to us nor their operators. We had to take action and invalidate the uptime's part of the formula used for calculating their reputation.
+
+We need to enable back uptime disqualification, however before, we need to track in a fair way the amount of time Storage Nodes are offline.
+
+This document presents a solution for tracking  Storage Node offline time, but it's out of the scope of this document how Storage Node reputation is affected by it.
+
+The presented solution considers the requirements imposed by the [Kademlia removal design document](kademia-removal.md), where the Storage Nodes must contact the Satellite when they start and after every hour.
+
+## Design
+
+For tracking Storage Nodes offline time we need:
+
+- A new SQL database table to store offline time.
+- Detecting which ones are offline.
+- Roughly estimate how long they are offline.
+
+An _uptime check_ referenced in this section is the connection initiated by the Satellite to any Storage Node in the same way that's described in [Network refreshing section of the Kademlia removal design document](kademia-removal.md#network-refreshing).
+
+__NOTE__ the SQL code in this section is illustrative for explaining the algorithm concisely.
+
+### Database
+
+The following new SQL database table will store Storage Nodes offline time.
+
+```sql
+CREATE TABLE nodes_offline_time(
+    node_id BYTEA NOT NULL,
+    tracked_at timestamp with time zone NOT NULL,
+    seconds integer,                                                    -- Measured number of seconds being offline
+    CONSTRAINT nodes_offline_time__pk PRIMARY KEY(node_id, tracked_at)
```

Comment:
do we need the constraint on `tracked_at`? It seems insufficient for detecting overlaps. At the same time, how likely is it to get the same timestamp.

Maybe this should be just an index.

Reply:
I'm thinking that we don't even need an index for the purpose of this document.

The document that should specify how to read this data it's the one that should specify which indexes are needed.

Reply:
I've removed the constraint.